spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/ChannelBindingServiceProperties.java

```diff
@@ -210,6 +181,13 @@ public ProducerProperties getProducerProperties(String outputChannelName) {
 
 	public BindingProperties getBindingProperties(String channelName) {
 		BindingProperties bindingProperties = new BindingProperties();
+		if (applicationContext != null) {
+			RelaxedDataBinder defaultsDataBinder = new RelaxedDataBinder(bindingProperties,
+					"spring.cloud.stream.default");
+			defaultsDataBinder.setDisallowedFields("destination");
+			defaultsDataBinder
+					.bind(new PropertySourcesPropertyValues(applicationContext.getEnvironment().getPropertySources()));
+		}
 		if (this.bindings.containsKey(channelName)) {
 			BeanUtils.copyProperties(this.bindings.get(channelName), bindingProperties);
```

Comment:
Doesn't this reset the values set by the dataBinder (with `spring.cloud.stream.default` prefix) above? While this is what we need to `override` the `spring.cloud.stream.default` values with the binding properties provided explicitly (via cmdline for instance), the `spring.cloud.stream.default` values will also get overridden with the application `default` values here.

For instance, if the application has `spring.cloud.stream.default.producer.partitionCount=5` that would never be set eventually because of BeanUtils.copyProperties at line:192.